kube/src/api/subresource.rs

```diff
@@ -221,6 +222,84 @@ where
     }
 }
 
+// ----------------------------------------------------------------------------
+// Eviction subresource
+// ----------------------------------------------------------------------------
+
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
+/// Eviction body
+#[derive(Default, Clone, Serialize)]
+pub struct Eviction {
+    /// How the eviction should occur
+    pub delete_options: Option<DeleteParams>,
+    /// Metadata describing the pod being evicted
+    /// we somehow need this despite providing the name of the pod..
+    pub metadata: ObjectMeta,
+}
+
+impl Eviction {
+    /// Create an eviction object for a named resource
+    pub fn new(name: &str) -> Self {
+        Self {
+            metadata: ObjectMeta {
+                name: Some(name.to_string()),
+                ..ObjectMeta::default()
+            },
+            delete_options: None,
+        }
+    }
+
+    /// Attach DeleteParams to the eviction object
+    pub fn with_options(mut self, dp: DeleteParams) -> Self {
+        self.delete_options = Some(dp);
+        self
+    }
+}
+
+impl Resource {
+    /// Create an eviction
+    pub fn evict(&self, name: &str, pp: &PostParams, data: Vec<u8>) -> Result<http::Request<Vec<u8>>> {
+        let base_url = self.make_url() + "/" + name + "/" + "eviction?";
+        // This is technically identical to Resource::create, but different url
+        pp.validate()?;
+        let mut qp = url::form_urlencoded::Serializer::new(base_url);
+        if pp.dry_run {
+            qp.append_pair("dryRun", "All");
+        }
+        let urlstr = qp.finish();
+        let req = http::Request::post(urlstr);
+        req.body(data).map_err(Error::HttpError)
+    }
+}
+
+#[test]
+fn evict_path() {
+    use crate::api::Resource;
+    use k8s_openapi::api::core::v1 as corev1;
+    let r = Resource::namespaced::<corev1::Pod>("ns");
+    let mut pp = PostParams::default();
+    pp.dry_run = true;
+    let req = r.evict("foo", &pp, vec![]).unwrap();
+    assert_eq!(req.uri(), "/api/v1/namespaces/ns/pods/foo/eviction?&dryRun=All");
+}
+
+/// Marker trait for objects that can be evicted
+pub trait EvictionObject {}
```

Comment:
Nit: `Evictable`, maybe? `EvictionObject` sounds like it's the reified object created when evicting something.

Reply:
Hm, I thought we had a convention, on noun-ing. But I am wrong.
Maybe we should just go full on on the conjugated form and rename all the traits?

i.e.
```rust
pub trait Loggable
pub trait Evictable
pub trait Executable
pub trait Atttachable
```

Reply:
I was thinking EvictableObject like AttachableObject.
Logging and Executing sounds fine to me, but not against making them consistent.
Naming is hard.

Reply:
Sounds good to me. I have been thinking about proposing that for *all* resource actions at some point (for example: `ComponentStatus` is read-only: https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#componentstatus-v1-core).

Reply:
I'll make a follow up PR for standardising. Sounds like we are OK with the conjugation as a convention.